command/agent/consul/int_test.go

```diff
@@ -136,6 +127,9 @@ func TestConsul_Integration(t *testing.T) {
 	if err := allocDir.Build(); err != nil {
 		t.Fatalf("error building alloc dir: %v", err)
 	}
+	t.Cleanup(func() {
+		r.NoError(allocDir.Destroy())
+	})
```

Comment:
The full test log: https://app.circleci.com/pipelines/github/hashicorp/nomad/23525/workflows/812a1d80-37b1-4251-8fc4-b7f192a7bf20/jobs/250947/tests#failed-test-0

At the middle of the logs, 
```
2022-05-03T07:40:32.419Z [WARN]  taskrunner/script_check_hook.go:381: task_runner.task_hook.script_checks: updating check failed: task=web error="Unexpected response code: 500 (CheckID \"_nomad-check-fe290e29ae5ee2da59e3bdbaf221c7b18e766e4f\" does not have associated TTL)"
    testing.go:967: TempDir RemoveAll cleanup: unlinkat /tmp/TestConsul_Integration1745651562/002/7f3f8cc1-961a-f83a-7daf-1554fd2cec19/web/secrets: device or resource busy
bootstrap = true: do not enable unless necessary
==> Starting Consul agent...
           Version: '1.8.3'
           Node ID: '2d6af774-8e9b-b8c8-cbe5-557349a513fc'
         Node name: 'node-2d6af774-8e9b-b8c8-cbe5-557349a513fc'
        Datacenter: 'dc1' (Segment: '<all>')
            Server: true (Bootstrap: true)
       Client Addr: [127.0.0.1] (HTTP: 22260, HTTPS: 22261, gRPC: -1, DNS: 22259)
      Cluster Addr: 127.0.0.1 (LAN: 22262, WAN: 22263)
           Encrypt: Gossip: false, TLS-Outgoing: false, TLS-Incoming: false, Auto-Encrypt-TLS: false

==> Log data will now stream in as it occurs:
```

`t.TempDir` failed to perform the cleanup because the child folder is still being used. `allocDir.Destroy()` is required here to unmount the directory

Reply:
Ok, thanks for following up.